sinaloa-test/src/main.rs

```diff
@@ -813,64 +860,230 @@ mod tests {
                     time_data.elapsed().as_micros()
                 );
             }
-            info!("### Step 7.  Result retrievers request program.");
-            let time_result_hash = Instant::now();
-            check_enclave_state(
-                client_session_id,
-                &mut client_session,
-                ticket,
-                &client_tls_tx,
-                &client_tls_rx,
-                ENCLAVE_STATE_READY_TO_EXECUTE,
-            )?;
-            let _response = request_program_hash(
-                policy.pi_hash().as_str(),
-                client_session_id,
-                &mut client_session,
-                ticket,
-                &client_tls_tx,
-                &client_tls_rx,
-            )?;
-            check_policy_hash(
-                &policy_hash,
-                client_session_id,
-                &mut client_session,
-                ticket,
-                &client_tls_tx,
-                &client_tls_rx,
-            )?;
-            info!(
-                "             Result retriever hash response time (μs): {}.",
-                time_result_hash.elapsed().as_micros()
-            );
-            let time_result = Instant::now();
-            info!("             Result retrievers request result.");
-            let response = client_tls_send(
-                &client_tls_tx,
-                &client_tls_rx,
-                client_session_id,
-                &mut client_session,
-                ticket,
-                &colima::serialize_request_result()?.as_slice(),
-            )
-            .and_then(|response| {
-                // decode the result
-                let response = colima::parse_mexico_city_response(&response)?;
-                let response = colima::parse_result(&response)?;
-                response.ok_or(SinaloaError::MissingFieldError(
-                    "Result retrievers response",
-                ))
-            })?;
-            info!(
-                "             Computation result time (μs): {}.",
-                time_result.elapsed().as_micros()
-            );
+            // If stream_id_pathes is NOT empty, we are in streaming mode
+            if !stream_id_pathes.is_empty() {
```

Comment:
Is streaming and the old batch-style of provisioning data now completely disjoint?  That is, is Veracruz either in a streaming mode or in a batch mode, and e.g. trying to provision batch-oriented data results in some error if in streaming mode?  If so, can we add tests to ensure that is the case?  If not, then what does the transition system look like, now, and do we need tests to check that Veracruz is progressing through the transition system as expected?

Reply:
I will add test cases for checking the state machine. In general, if a participant tries to provision stream data in a wrong state, it will raise an error, similar to that if a participant provisions data before there is a program.